### PR TITLE
Release version 1.0.0

### PR DESCRIPTION
**Pull request by dfalster**
First major release of APCalign.  A preprint is available at 
https://www.biorxiv.org/content/10.1101/2024.02.02.578715v1. 
Article has been accepted for publication at Australian Journal of Botany.

Following review, a number of changes have been implemented. These have sped & 
streamlined the package.

* Update function documentation
* Speed up `extract_genus`
* Write a replacement function for `stringr::word` that is much faster.
* Additional speed up and accuracy of `fuzzy_match` function by 
  - Restricting reference list to names with the same first letter as input string.
  - Switch from using `utils::adist` to `stringdist:stringdist(method = "dl")`
* Rework `standardise_names` to remove punctuation from the start of the string
* Rework `strip_names_extra` (previously `strip_names_2`) to just perform 
additional functions to `strip_names`, rather than repeating those performed by `strip_names`.
* Avoid importing entire packages by using package::function format throughout 
and removing functions from @import
* Add fuzzy match arguments to `create_taxonomic_update_lookup`
* Add 3 additional family-level APC matches to `match_taxa`.
* Refine tests
* Make messages to console optional
* Fix issue with fails when github is down (https://github.com/traitecoevo/APCalign/issues/205)

### CODE REVIEW

**Review by ehwenk — APPROVED** [review]
It is excellent to see how small refinements can keep improving the package.

I corrected a few typos in your message, but otherwise all good.

File: R/reexports.R
Comment (fontikar) [inline R/reexports.R:1]: I think the `%>%` is from `magrittr`? can use `devtools::use_pipe` to import and document this :)

**Review by fontikar — DISMISSED** [review]
Minor change for importing pipe but not crucial!

File: R/reexports.R
Comment (dfalster) [reply R/reexports.R:1]: Yes, but I decided to use dplyr version to reduce dependency load

**Pull request merged**